resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -173,10 +183,12 @@ private[spark] class Client(
         .bootstrapInitContainerAndVolumes(driverContainer.getName, basePod)
 
     val containerLocalizedFilesResolver = initContainerComponentsProvider
-        .provideContainerLocalizedFilesResolver()
+        .provideContainerLocalizedFilesResolver(mainAppResource)
     val resolvedSparkJars = containerLocalizedFilesResolver.resolveSubmittedSparkJars()
     val resolvedSparkFiles = containerLocalizedFilesResolver.resolveSubmittedSparkFiles()
-
+    val resolvedPySparkFiles = containerLocalizedFilesResolver.resolveSubmittedPySparkFiles()
+    val resolvedPrimaryPySparkResource = if (!isPython) ""
+     else { containerLocalizedFilesResolver.resolvePrimaryResourceFile() }
```

Comment:
Braces should always follow

```
if (...) {
   // logic
} else {
   // other logic
}
```

regardless of how long the logic in each block is. The exception to this rule is for something that should act like a ternary operator and can fit entirely in the same line.